setuptools/tests/test_build_meta.py

```diff
@@ -440,18 +595,28 @@ def run():
     def test_sys_argv_passthrough(self, tmpdir_cwd):
         path.build(self._sys_argv_0_passthrough)
         build_backend = self.get_build_backend()
-        with pytest.raises(AssertionError):
-            build_backend.build_sdist("temp")
+
+        # TODO: Clarify the use case motivating this test
+        # Why is it necessary that the assertion error is raised?
+        # Or is it just a side effect that was found to happen,
+        # but it is not necessary for setuptools to work properly?
+        # >>> with pytest.raises(AssertionError):
+
+        build_backend.build_sdist("temp")
 
     @pytest.mark.parametrize('build_hook', ('build_sdist', 'build_wheel'))
     def test_build_with_empty_setuppy(self, build_backend, build_hook):
         files = {'setup.py': ''}
         path.build(files)
 
-        with pytest.raises(
-                ValueError,
-                match=re.escape('No distribution was found.')):
```

Comment:
Is this a desired behaviour, or is it acceptable that setuptools simply proceed if the `setup.py` script is empty?

Reply:
(The reason for this question is that the code in the PR is simply creating a `Distribution` object with the default parameters when the `setup.py` file is empty, but that can be changed...)

Reply:
I'd like to get away from supporting setup.py, but if a setup.py is present, there's a reasonable expectation that it will be invoked and that it must invoke `setuptools.setup()` when executed with `__name__ == '__main__'`. So yes, an empty setup.py should be invalid.

It may make sense to loosen this expectation.

Reply:
In https://github.com/pypa/setuptools/pull/3064 I decided to explicitly fail if the setup script is empty to avoid subtle accidental behaviour.

In the exception message I am also trying to emphasize that setuptools no longer need `setup.py`. Hopefully this will incentivise users to move away from it. See https://github.com/pypa/setuptools/pull/3064/commits/660a273f67dc0aa3e9d2818079051a4b927d91bf.

Reply:
https://github.com/pypa/setuptools/commit/660a273f67dc0aa3e9d2818079051a4b927d91bf was actually reverted. Now the behaviour of `build_meta` is preserved.